test/Microsoft.ML.Functional.Tests/Common.cs

```diff
@@ -160,6 +161,73 @@ public static void AssertEqual(TypeTestData testType1, TypeTestData testType2)
             Assert.True(testType1.Ug.Equals(testType2.Ug));
         }
 
+        /// <summary>
+        /// Check that a <see cref="AnomalyDetectionMetrics"/> object is valid.
+        /// </summary>
+        /// <param name="metrics">The metrics object.</param>
+        public static void CheckMetrics(AnomalyDetectionMetrics metrics)
+        {
+            // Perform sanity checks on the metrics.
+            Assert.InRange(metrics.Auc, 0, 1);
+            Assert.InRange(metrics.DrAtK, 0, 1);
+        }
+
+        /// <summary>
+        /// Check that a <see cref="BinaryClassificationMetrics"/> object is valid.
+        /// </summary>
+        /// <param name="metrics">The metrics object.</param>
+        public static void CheckMetrics(BinaryClassificationMetrics metrics)
```

Comment:
>BinaryClassificationMetrics [](start = 40, length = 27)

we added `CalibratedBinaryClassificationMetrics` recently. Strangely you don't have them in your PR.

Reply:
Good catch. I hadn't noticed that.

---
In reply to: [259454652](https://github.com/dotnet/machinelearning/pull/2646#discussion_r259454652) [](ancestors = 259454652)